### DISCUSSION TITLE
Comparing Time Series Model Performance | Managing Time Constraints

### DISCUSSION THREAD

**Discussion by ngupta23** [pycaret/pycaret#1809] [FAQs]
# Background

In a **[previous FAQ](https://github.com/pycaret/pycaret/discussions/1808)**, we saw how easy it is to consistently compare the performance of multiple time series models in `pycaret`. However, building so many models can be time consuming, especially for large datasets. Sometimes, users may want to limit the run based on a time allowance. This is especially helpful when you have to scale to 1000's of time series but only have a limited time to create the models. On other occasions, models that train slower may lead to better performance for a user's dataset and so the user may be willing to take the time hit to get a better model. Luckily, `pycaret` provides the user all these capabilities. Let's see how.

NOTE: A complete notebook example for comparing models can be **[found here](https://gist.github.com/ngupta23/f94b2f3ce22b7a0203446b77f2972ffa)**.

# Managing Time Constraints

## Enabling slower models

By default, compare_models will only run fast models. Slower models such as `bats`, `tbats`, `prophet` are not run by default. We can enable these models by setting `turbo=False` in the `compare_models` call.

```python
all_baseline_inc_slow = exp.compare_models(turbo=False)
```
![image](https://user-images.githubusercontent.com/33585645/140610380-fc74d985-3ec6-453f-8ba2-240afc306873.png)

## Run time budget

As mentioned above, sometimes, users may want to limit the run based on a budgeted time. This can be done easily using the `budget_time` argument which limits the run to the specified number of minutes. Once that budget is elapsed, the `compare_model` run is stopped and only models that have been trained in that time frame are displayed/returned.

```python
all_baseline_some_models = exp.compare_models(turbo=False, budget_time=0.1)
```
![image](https://user-images.githubusercontent.com/33585645/140610434-9fd2f7b3-d4dd-4fd7-a4da-37061588d727.png)

That is how easy it is to customize time allowance and potential performance tradeoffs in `pycaret`. That's it for this FAQ. Be sure to read the suggested next reads to see how you can customize the comparison process further to suit your specific needs.

Happy Forecasting!

# Suggested Next Reads

- https://github.com/pycaret/pycaret/discussions/1810

**Comment by moezali1** [comment]
PyCaret 4.0 sunset the legacy time-series module (sktime/prophet integration), so this FAQ no longer maps to a supported code path.

If you still need time-series forecasting, use sktime / prophet / statsforecast directly — those projects have surpassed what the PyCaret wrapper offered.

For 4.0 docs, see https://pycaret.org/docs and the cleanup notice at https://github.com/pycaret/pycaret/issues/4177.

Closing as outdated.
